src/flag_gems/utils/shape_utils.py

```diff
@@ -137,3 +139,30 @@ def dim_compress(inp, dims):
     sorted_reduction_dim = sorted(dims, key=lambda x: stride[x], reverse=True)
     order = batch_dim + sorted_reduction_dim
     return inp.permute(order).contiguous()
+
+
+def offsetCalculator(inp, idx, strides, dim, isInp):
```

Comment:
Can we doc this function to help others understand it?

Reply:
The offset calculation incurs massive overhead. Let's try to do it in a Triton kernel, shall we?